website/docs/r/postgresql_virtual_network_rule.html.markdown

```diff
@@ -0,0 +1,99 @@
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-postgresql-server-vnet-rule"
+  location = "West US"
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "example-vnet"
+  address_space       = ["10.7.29.0/29"]
+  location            = "${azurerm_resource_group.example.location}"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+}
+
+resource "azurerm_subnet" "subnet" {
+  name                 = "example-subnet"
+  resource_group_name  = "${azurerm_resource_group.example.name}"
+  virtual_network_name = "${azurerm_virtual_network.vnet.name}"
+  address_prefix       = "10.7.29.0/29"
+  service_endpoints    = ["Microsoft.Sql"]
+}
+
+resource "azurerm_postgresql_server" "postgresql_server" {
+  name                = "postgresql-server-1"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  sku {
+    name = "B_Gen4_2"
+    capacity = 2
+    tier = "Basic"
+    family = "Gen4"
+  }
+
+  storage_profile {
+    storage_mb = 5120
+    backup_retention_days = 7
+    geo_redundant_backup = "Disabled"
+  }
+
+  administrator_login = "psqladminun"
+  administrator_login_password = "H@Sh1CoR3!"
+  version = "9.5"
+  ssl_enforcement = "Enabled"
+}
+
+resource "azurerm_postgresql_virtual_network_rule" "postgresqlsql_virtual_network_rule" {
+  name                = "postgresql-vnet-rule"
+  resource_group_name = "${azurerm_resource_group.example.name}"
+  server_name         = "${azurerm_postgresql_server.postgresql_server.name}"
+  subnet_id           = "${azurerm_subnet.subnet.id}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the SQL virtual network rule. Cannot be empty and must only contain alphanumeric characters and hyphens. Cannot start with a number, and cannot start or end with a hyphen. Changing this forces a new resource to be created.
+
+~> **NOTE:** `name` must be between 1-128 characters long and must satisfy all of the requirements below:
+1. Contains only alphanumeric and hyphen characters
+2. Cannot start with a number or hyphen
+3. Cannot end with a hyphen
+
+* `resource_group_name` - (Required) The name of the resource group where the PostgreSQL server resides. Changing this forces a new resource to be created.
+
+* `server_name` - (Required) The name of the SQL Server to which this PostgreSQL virtual network rule will be applied to. Changing this forces a new resource to be created.
+
+* `subnet_id` - (Required) The ID of the subnet that the PostgreSQL server will be connected to.
+
+~> **NOTE:** The resource is configured with `ignore_missing_vnet_service_endpoint` set to `true`, meaning the deployment will succeed even if the target subnet does not contain the `Microsoft.Sql` endpoint in the `service_endpoints` array. This attribute will be introduced once the API behaviour is consistent.
```

Comment:
given this defaults to `false` in the other resources, I think it'd be sensible to make this `false` by default for the moment (since otherwise this'll cause conflicts for users when we introduce this field with a default of `false` when it's fixed) - what do you think?

Reply:
I had a brief debate about this with myself and what happens when the endpoint is missing.

I see where you're coming from and perhaps it would be best to do that. I was going to ask what you think is the best approach from a user experience perspective.

* If `IgnoreMissingVnetServiceEndpoint` is false and the endpoint is missing, the resource creation will timeout.
* If `IgnoreMissingVnetServiceEndpoint` is true and the endpoing is missing, the resource creation will succeed but the functionality will be broken.

I suspect your suggestion is probably the better choice but I'm curious as to what you think.

Reply:
Oh.

Reply:
whilst we've removed this, to ensure this fails gracefully I've pushed a commit (I hope you don't mind!) to add the above code to the Create method, to ensure the Subnet's in a usable state prior to creating the subnet rule